### DISCUSSION TITLE
Sample IKEv1 Configuration

### DISCUSSION THREAD

**Discussion by jhay06** [strongswan/strongswan#1062] [Q&A]
Hi would like to request a sample client configuration , i have no to zero knowledge on the strongswan
actually i don't know who is the left and right participant ,

my target is to connect to my company network using linux

example the company domain network is example.com port 500
my company provides signed certificate .crt file and a private key .prv file
my company also provide .ca cert file 
The company use ikev1
The company phase 1 algo is 3des-sha1 
The company phase 2 algo is also 3des-sha1
The company assigned the client virtual ip into 10.2.0.0/20

I have the generated .csr file 
I have the .crt file and the .prv file that the company provided 
i have the login identity username@example.com
my ip address for example is 192.168.2.5

what would be the configuration for me to connect to the company? 

Thanks

**Comment by tobiasbrunner** [comment]
You might want to start [here](https://docs.strongswan.org/docs/5.9/index.html).

**Reply by jhay06** [reply]
hi . thanks for the response . if you dont mind . can you attach sample config base on the sample scenario if it is okay?thanks

**Reply by tobiasbrunner** [reply]
There is a link called **quickstart examples** on that page that provides sample configs.

**Reply by jhay06** [reply]
actually i already tried swanctl and ipsec.conf , but no luck in swanctl it doesn't connect at all , in strongswan + ipsec.conf it doesn't connect on 2nd round because i dont know if my config is correct or not . that's why i'm asking for a sample config base on the sample scenario . for me to undestand what's wrong on the config 

strongswan + ipsec 

conn sample
  type=tunnel
  auto=start
  keyexchange=ikev1
  leftauth=pubkey
  leftauth2=xauth
  leftcert=mysignedcert.crt
  leftid="i dont know what to add this , but i followed some of the config from the internet , i put me@companydomain.com"
  xauth_identity=same as leftid
  right="ip of the network of the company"
  rightid="i dont know what to add this but i followed some of the config from the internet , i put %{ip of the company here}
  rightauth=pubkey
  rightcert=companypub.ca
  rightcert2=companypub2.ca
  ike=3des-sha1-modp1024
  esp=3des-sha1-modp1024
  authby=secret

for the ipsec.secrets 
: RSA myprivatekey.prv
me@companydomain.com : XAUTH "passkey"

 it pass on 1st auth but doesnt continue on second auth

for swanctl .conf


connections {
  mycon {
    version = 1 # i this is the version of ike used so i put 1
    fragmentation = yes
    # I think this is the set of ciphersuites available for IKE?
    proposals = 3des-sha1-modp1024
    aggressive = no

    remote_addrs= "i assumed this is the company ip address , so i put the company host/ip address to connect to"
 
    local {
  
      certs = mysignedcert.crt
    
      auth = xauth
      
      xauth_id =me@companydomain.com
    }
    
      remote {
      auth =pubkey
      cacerts =  companypub.ca
    }
    

    children {
    # idont what to here, but i just copy it from the internet and edit some parameter
      myconchild {
        start_action = trap
        close_action = trap

        esp_proposals = 3des-sha1-modp1024
        mode = transport
        
       #i dont what to expect here if this is the same as on the charon-cmd
        local_ts = 172.28.0.0/20
        remote_ts = 0.0.0.0/0
        
   
        dpd_action = restart
      }
    }
  }
}


but as soon as i initiate  this config , it doesn't connect to IP of the company ,says connection aborted .

**Reply by tobiasbrunner** [reply]
`--profile ikev1-pub` clearly has nothing to do with XAuth but uses `pubkey` authentication for both ends.

> it doesn't connect to IP of the company ,says connection aborted .

It would hep if you posted (or even read) the log (see #196).

**Reply by jhay06** [reply]
Hi i figured out whats the problem here, i think charon-cmd does not support multiple subnets or ip's 

on our company we have different sites with different subnets 

example 
192.168.XXX.XXX/24
10.XXX.XXX.XXX/16
192.168.XXX.101-192.168.XXX.255 

how can i achieve the multiple remote ts ? 

Thanks

**Reply by tobiasbrunner** [reply]
charon-cmd does support multiple traffic selectors/subnets (`--remote-ts` can be passed multiple times), **but not when using IKEv1** because that requires separate CHILD_SAs (see [this FAQ entry](https://docs.strongswan.org/docs/5.9/support/faq.html#_multiple_subnets_per_sa)). That's the same with swanctl.conf, where you have to add separate sections in `children` for each combination of local and remote traffic selectors (whereas with IKEv2 you could just add multiple, comma-separated subnets in `remote_ts`).

**Reply by jhay06** [reply]
any other way sir ? i really need this , so i can stop using windows to connect to our server.

Thanks

**Comment by jhay06** [comment]
any update ? 

Please help me , i tried many times but its failing 

i just want to convert this command from charon-cmd to swanctl or ipsec.conf 

sudo charon-cmd --host my.company.ip --identity me@InternalDomain
--ike-proposal 3des-sha1-modp1024 --esp-proposal 3des-sha1-modp1024 --xauth-username me@InternalDomain
--cert mysigned.crt --rsa mykey.prv --profile ikev1-pub
--cert mycompany1pub.ca --cert mycompany2pub.ca

**Reply by jhay06** [reply]
hi @tobiasbrunner , can you help me how to convert this charon cmd to ipsec.conf ? 

Thanks 

i tried many times but i cannot make it work,
i tried libreswan and i can successfully connect but the same behavior with charon cmd only 1 subnet can be added , i found out that strongswan supports multiple subnet (remote ts) so i prefer to use strongswan ,

**Comment by jhay06** [comment]
Hi , im hoping that someone can update charon-cmd to allow ikev1 to use mulitple subnets , only charon-cmd is currently working , strongswan ipsec.conf and swanctl.conf doesn't work.

**Reply by tobiasbrunner** [reply]
> im hoping that someone can update charon-cmd to allow ikev1 to use mulitple subnets

That's very unlikely to happen. 

> only charon-cmd is currently working , strongswan ipsec.conf and swanctl.conf doesn't work.

You just have to get the config right. `charon-cmd` doesn't do anything you can't configure with these other config backends.
